pkg/resource/deploy/step_generator.go

```diff
@@ -330,28 +331,30 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, *re
 					// we must also eagerly delete all resources that depend directly or indirectly on the resource
 					// being replaced.
 					//
-					// To do this, we'll utilize the dependency information contained in the snapshot, which is
-					// interpreted by the DependencyGraph type.
+					// To do this, we'll utilize the dependency information contained in the snapshot if it is
+					// trustworthy, which is interpreted by the DependencyGraph type.
 					var steps []Step
-					dependents := sg.plan.depGraph.DependingOn(old)
-
-					// Deletions must occur in reverse dependency order, and `deps` is returned in dependency
-					// order, so we iterate in reverse.
-					for i := len(dependents) - 1; i >= 0; i-- {
-						dependentResource := dependents[i]
-
-						// If we already deleted this resource due to some other DBR, don't do it again.
-						if sg.deletes[urn] {
-							continue
+					if sg.opts.TrustDependencies {
```

Comment:
Just to be clear, this returns us to the original behavior of delete-before-create when we're running a Python program, right? Are Python dependencies incomplete, or just wrong?

Reply:
> Just to be clear, this returns us to the original behavior of delete-before-create when we're running a Python program, right?

Yes, that's correct.

> Are Python dependencies incomplete, or just wrong?

They're incomplete. Python does not honor `dependsOn` nor does it make any attempt to track implicit dependencies.

Reply:
To be clear, there's most likely going to be no behavior change here when using Python. The Python dependency graph is always empty because no dependencies are reported. When adding `TrustDependencies`, I figured that it should guard against any use of dependency graphs, since it's (in my mind) wrong to use one if we know that the information contained in it is incomplete and untrustworthy.

Reply:
FWIW, we could "improve" this situation by handling this in the language host and having it report all preceding resources as dependencies. I don't think that we _should_ do that--there are few enough places where we use dependency information that I don't believe that it's necessary--but I think that it's worth mentioning for completeness.